### PR TITLE
add readme + move output into dist/

### PR DESCRIPTION
**Pull request by chancesmith**
## overall purpose
So developer knows where to look for generated output files, what if we put them all in dist/?

## changes made
- put zip into dist/
- put signatures into dist/signatures/
- put report into dist/signatures/

### CODE REVIEW

File: csv-bulk-signature/.gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
 node_modules/
 dist/
 contacts.csv
-signatures-*.zip
```
Comment (chancesmith) [inline csv-bulk-signature/.gitignore:4]: zip is now in dist/

File: csv-bulk-signature/README.md [hunk 2]
```diff
@@ -5,3 +5,10 @@ Process CSV file to build/download multiple email signatures to local computer a
 You'll need to clone and add CSV to src/ then run gulp and look inside dist/index.html to export a ton of signatures. :)
 
 ![bulk-email-signature-generator](http://sodiumhalogen.com/up_c/bulk-download-Ta6vxKqPf6.gif)
+
+## steps to run the script
+
+1. get a list of contacts into a `src/contacts.csv`
+2. in this folder run: `yarn` and then ` yarn bulk` or if you are developing `y bulk:watch`
+3. expect to see a `./dist/signatures` folder show all the signatures
+4. expect to see a `./dist/signature-DATE.zip` that holds a `_statusReport.txt` and every email signature.
```
Comment (chancesmith) [inline csv-bulk-signature/README.md:14]: setup instructions

File: csv-bulk-signature/package.json [hunk 3]
```diff
@@ -19,6 +19,9 @@
     "handlebars": "^4.7.7",
     "inline-css": "^4.0.1",
     "jszip": "^3.10.1",
-    "nodemon": "^2.0.20"
+    "nodemon": "^2.0.20",
+    "ts": "^0.2.2",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.9.4"
```
Comment (chancesmith) [inline csv-bulk-signature/package.json:25]: add deps for local dev setup

File: csv-bulk-signature/src/_example-contacts.csv [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Brand*,Full Name*,Credentials,Title*,Office Phone*,Mobile Phone,Calendly Link
+use dropdown to select a brand,ex. Joe Smith,"ex. CPA, CVGA, CGMA",ex. Marketing Director,ex. 731.285.7900,ex. 731.285.7900,ex. https://calendly.com/shjeremy
+image-id,First Last,"CPA, CFE",Chief Manager ,777.555.7777,777.444.5555,https://calendly.com/EXAMPLE
```
Comment (chancesmith) [inline csv-bulk-signature/src/_example-contacts.csv:3]: add example for contacts so dev knows where to put CSV

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -41,6 +41,14 @@ interface TemplateData {
 const skippedRows: string[] = [];
 const processedRows: string[] = [];
 
+// create ./dist/signatures if it doesn't exist
+if (!fs.existsSync("./dist")) {
+  fs.mkdirSync("./dist");
+}
+if (!fs.existsSync("./dist/signatures")) {
+  fs.mkdirSync("./dist/signatures");
+}
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:50]: add missing folders if they don't exist

File: csv-bulk-signature/src/main.ts [hunk 6]
```diff
@@ -63,6 +71,9 @@ function generateSignatures(
   fs.createReadStream(CSV_FILE)
     .pipe(csv())
     .on("data", (row: Contact) => {
+      // @ts-ignore - this is a hack to skip the first row
+      if (row["Brand*"] === "use dropdown to select a brand") return;
+
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:76]: skip first row of csv that has example data

File: csv-bulk-signature/src/main.ts [hunk 7]
```diff
@@ -89,7 +100,7 @@ function generateSignatures(
           calendly,
         });
         fs.writeFile(
-          `./dist/${fullName.split(" ").join("")}.htm`,
+          `./dist/signatures/${fullName.split(" ").join("")}.htm`,
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:103]: put all signatures into dist/signatures/ so zips can live in root of dist/

File: csv-bulk-signature/src/main.ts [hunk 8]
```diff
@@ -102,8 +113,13 @@ function generateSignatures(
...
       console.log("Number of signatures skipped", skippedRows.length);
       createStatusReport(); // refactoring the results specifics to be shown in a text file, instead
+
+      // wait three seconds before zipping up the files
+      console.log("Zipping up files...");
+      setTimeout(() => {
+        zipUpFile();
+      }, 1500);
     });
-  zipUpFile();
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:106]: don't zip until we have all signatures created --- hack by waiting a couple seconds

File: csv-bulk-signature/src/main.ts [hunk 9]
```diff
@@ -162,18 +178,20 @@ function checkHeadersToBeSame() {
         data.hasOwnProperty(expectedHeader)
       );
       if (headersMatch) {
-        console.log("Headers match ::thumbs::");
+        console.log("CSV Headers match 👍");
       } else {
-        console.log("Headers do not match ::warning::");
+        throw new Error("CSV Headers do not match 🙅‍♂️");
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:183]: update logs and error if headers aren't happy

File: csv-bulk-signature/src/main.ts [hunk 10]
```diff
@@ -162,18 +178,20 @@ function checkHeadersToBeSame() {
...
       }
       counter = 1;
     });
 }
 
 async function zipUpFile() {
-  const folderPath = "./dist";
+  const folderPath = "./dist/signatures";
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:190]: get signatures from dist/signatures/

File: csv-bulk-signature/src/main.ts [hunk 11]
```diff
@@ -162,18 +178,20 @@ function checkHeadersToBeSame() {
...
   const files = fs.readdirSync(folderPath);
-  const filesHtm = files.filter((file) => file.endsWith(".htm"));
+  const filesHtm = files.filter(
+    (file) => file.endsWith(".htm") || file.endsWith(".txt")
+  );
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:194]: look for report file too

File: csv-bulk-signature/src/main.ts [hunk 12]
```diff
@@ -182,24 +200,30 @@ async function zipUpFile() {
...
   });
 
   const buffer = await zip.generateAsync({ type: "nodebuffer" });
   fs.writeFileSync(
-    `signatures-${new Date().toISOString().split("T")[0]}.zip`,
+    `./dist/signatures-${
+      new Date().toISOString().split(":").join("_").split(".")[0] // date and time
+    }.zip`,
     buffer
   );
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:214]: so the client and dev knows which zip is which, included time stamp in zip filename too

File: csv-bulk-signature/src/main.ts [hunk 13]
```diff
@@ -182,24 +200,30 @@ async function zipUpFile() {
...
-  });
+  const combinedStatus = skippedRows.concat(processedRows);
+  fs.writeFile(
+    "./dist/signatures/_statusReport.txt",
+    combinedStatus.join("\n"),
+    (err) => {
+      if (err) throw err;
+      console.log("Report created 👍");
+    }
+  );
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts:228]: move report into signatures folder and sort to the top with the underscore

**Review by adamnitti — APPROVED** [review]

**Pull request merged**